tests/test_socket_server.py

```diff
@@ -46,21 +47,20 @@ async def asyncSetUp(self):
 
         self.log = logging.getLogger(type(self).__name__)
 
-        self.assertFalse(self.srv._started)
-        asyncio.create_task(self.srv.start())
-        await asyncio.sleep(0.5)
-        self.assertTrue(self.srv._started)
+        await self.srv.start()
         # Request the assigned port from the mock controller.
         port = self.srv.port
 
-        rw_coro = asyncio.open_connection(host="127.0.0.1", port=port)
-        self.reader, self.writer = await asyncio.wait_for(rw_coro, timeout=TIMEOUT)
+        logging.info(f"Connecting to port {port}")
+        self.reader, self.writer = await asyncio.open_connection(
+            host="127.0.0.1", port=port
```

Comment:
Consider `host=tcpip.LOCAL_HOST`

Also, do you use `port = self.src.port at line 52 anywhere but here? Consider `port=self.srv.port` here, instead.

Reply:
Done